knowledge-base/autocomplete-tokens-mvvm.md

```diff
@@ -0,0 +1,217 @@
+    {
+        if (source == null || target == null)
+            return;
+
+        target.Clear();
+        foreach (var o in source)
+        {
+            target.Add(o);
+        }
+    }
+}
+```
+
+**2.** Set the behavior to the AutoComplete control:
+
+```XAML
+<VerticalStackLayout>
+    <telerik:RadAutoComplete x:Name="autoCompleteView"
+                                        ItemsSource="{Binding Source}"
+                                        TextSearchPath="Name" 
+                                        DisplayMode="Tokens">
+        <telerik:RadAutoComplete.Behaviors>
+            <local:TokensBehavior BindingContext="{Binding Source={x:Reference autoCompleteView}, Path=BindingContext}"
+                                    Tokens="{Binding SelectedTokens}"/>
+        </telerik:RadAutoComplete.Behaviors>
+    </telerik:RadAutoComplete>
+    <Button Text="Select Token" Command="{Binding SelectTokenCommand}"/>
+    <Button Text="Remove Token" Command="{Binding RemoveTokenCommand}"/>
+</VerticalStackLayout>
+```
+
+**3.** Add the `telerik` namespace:
+
+```XAML
+xmlns:telerik="http://schemas.telerik.com/2022/xaml/maui"
+```
+
+**4.** Add sample model:
+
+```C#
+public class Customer
+{
+    public Customer(string name, string email)
+    {
+        this.Name = name;
+        this.Email = email;
+    }
+    public string Name { get; set; }
+    public string Email { get; set; }
+}
+```
+
+**5.** Add the `ViewModel`:
+
+```C#
+public class ViewModel
+{
+    public ObservableCollection<Customer> Source { get; set; }
+    public ViewModel()
+    {
+        this.Source = new ObservableCollection<Customer>()
+            {
+                new Customer("Freda Curtis", "fcurtis@mail.com"),
+                new Customer("Jeffery Francis", "jfrancis@mail.com"),
+                new Customer("Eva Lawson", "elawson@mail.com"),
+                new Customer("Emmett Santos", "esantos@mail.com"),
+                new Customer("Theresa Bryan", "tbryan@mail.com"),
+                new Customer("Jenny Fuller", "jfuller@mail.com"),
+                new Customer("Terrell Norris", "tnorris@mail.com"),
+            };
+
+        this.SelectTokenCommand = new Command(OnSelectTokenCommandExecuted);
+        this.RemoveTokenCommand = new Command(OnRemoveTokenCommandExecuted);
+
+        this.SelectedTokens = new ObservableCollection<object>();
+    }
+
+    public ICommand SelectTokenCommand { get; set; }
+    public ICommand RemoveTokenCommand { get; set; }
+
+    public ObservableCollection<object> SelectedTokens { get; set; }
+
+    private void OnSelectTokenCommandExecuted(object obj)
+    {
+        this.SelectedTokens.Add(this.Source[0]);
+    }
+
+    private void OnRemoveTokenCommandExecuted(object obj)
+    {
+        var item = this.SelectedTokens.FirstOrDefault();
+        if (item != null)
+        {
+            this.SelectedTokens.Remove(item);
+        }
+    }
+}
+```
```

Comment:
Add a See Also/Related Articles section containing links to two or three relevant articles, e.g. https://docs.telerik.com/devtools/maui/controls/autocomplete/tokens-support